src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1897,6 +1897,8 @@ FilterJoinRestrictionContext(JoinRestrictionContext *joinRestrictionContext, Rel
 		}
 	}
 
+	filtererdJoinRestrictionContext->hasSemiJoin = joinRestrictionContext->hasSemiJoin;
```

Comment:
maybe add  the following comment:
`/* the filtered restriction might not have semiJoin, but it is OK for now */ `

Reply:
or, maybe build `hasSemiJoin ` on the filtered ones